### PR TITLE
Feature/updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
- Add CircleCi integration tests
- Staging table set up
- Include all default columns in table
- Create Pass through column variables
- Update ReadMe and add Netlify docs site
- Added comments for easier understanding of complexity in models. See below for example:
```
macro as (
    select
        /*
        The below macro is used to generate the correct SQL for package staging models. It takes a list of columns 
        that are expected/needed (staging_columns from dbt_github_source/models/tmp/) and compares it with columns 
        in the source (source_columns from dbt_github_source/macros/).

        For more information refer to our dbt_fivetran_utils documentation (https://github.com/fivetran/dbt_fivetran_utils.git).
        */
            {{
            fivetran_utils.fill_staging_columns(
                source_columns=adapter.get_columns_in_relation(ref('stg_github__pull_request_tmp')),
                staging_columns=get_pull_request_columns()
            )
        }}

        --The below script allows for pass through columns.
        {% if var('pull_request_pass_through_columns') != [] %}
        ,
        {{ var('pull_request_pass_through_columns') | join (", ")}}

        {% endif %}
    from pull_request
```

- Included `issue_comment_id` and `pull_request_review_id` (not previously included in stage models) in respective models to allow for tests to still be applied in stg_github.yml

### CODE REVIEW

**Comment by fivetran-joemarkiewicz** [comment]
I noticed **Contributions** section in ReadMe is duplicated. Putting a reminder in for myself to fix.

File: dbt_project.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
 config-version: 2
 
 name: 'github_source'
-version: '0.0.1'
+version: '1.0.0'
```
Comment (fivetran-jamie) [inline dbt_project.yml:4]: should be 0.1.0 🤠

File: integration_tests/ci/sample.profiles.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+
+# HEY! This file is used in the Intercom integrations tests with CircleCI.
```
Comment (fivetran-jamie) [inline integration_tests/ci/sample.profiles.yml:2]: intercom should be replaced w github

File: integration_tests/dbt_project.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: 'github_source_integration_tests'
+version: '1.0.0'
```
Comment (fivetran-jamie) [inline integration_tests/dbt_project.yml:2]: 0.1.0 probably

File: models/stg_github.yml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        description: Foreign key that references the issue table
+      - name: user_id
+        description: Foreign key that references the user table
+
+  - name: stg_github__issue_merged
+    description: Table that tracks when an issue (always a pull request) is merged
+    columns:
+      - name: issue_id
+        description: Foreign key that references the issue table.  This table will only refernece issues that are pull requests
+      - name: maerged_at
```
Comment (fivetran-jamie) [inline models/stg_github.yml:57]: should this be `merged_at`?

File: models/stg_github__issue.sql [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      locked as is_locked,
+      milestone_id,
+      number as issue_number,
+      pull_request as is_pull_request,
+      repository_id,
+      state,
+      title,
+      updated_at,
+      user_id
+    from macro
```
Comment (fivetran-jamie) [inline models/stg_github__issue.sql:46]: ah gotta include the passthrough column stuff here as well @fivetran-joemarkiewicz

File: models/stg_github__issue_assignee.sql [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+        {% endif %}
+    from issue_assignee
+
+), fields as (
+
+    select 
+      issue_id,
+      user_id
+    from macro
```
Comment (fivetran-jamie) [inline models/stg_github__issue_assignee.sql:35]: need passthrough columns

File: models/stg_github__issue_closed_history.sql [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        {% endif %}
+    from issue_closed_history
+
+), fields as (
+
+    select 
+      issue_id,
+      updated_at,
+      closed as is_closed
+    from macro
```
Comment (fivetran-jamie) [inline models/stg_github__issue_closed_history.sql:36]: passthru

File: models/stg_github__issue_comment.sql [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+        {% endif %}
+    from issue_comment
+
+), fields as (
+
+    select 
+      id as issue_comment_id,
+      issue_id,
+      user_id
```
Comment (fivetran-jamie) [inline models/stg_github__issue_comment.sql:35]: passthru

File: models/stg_github__issue_label.sql [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+        {% endif %}
+    from issue_label
+
+), fields as (
+
+    select 
+      issue_id,
+      label
+    from macro
```
Comment (fivetran-jamie) [inline models/stg_github__issue_label.sql:35]: passthru

File: models/stg_github__issue_merged.sql [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        {{ var('issue_merged_pass_through_columns') | join (", ")}}
+
+        {% endif %}
+    from issue_merged
+
+), fields as (
+
+    select 
+      issue_id,
+      merged_at
```
Comment (fivetran-jamie) [inline models/stg_github__issue_merged.sql:34]: passthru

File: models/stg_github__pull_request.sql [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        {% endif %}
+    from pull_request
+
+), fields as (
+
+    select 
+      id as pull_request_id,
+      issue_id,
+      head_repo_id,
+      head_user_id
```
Comment (fivetran-jamie) [inline models/stg_github__pull_request.sql:36]: passthru

File: models/stg_github__pull_request_review.sql [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+), fields as (
+
+    select 
+      id as pull_request_review_id,
+      pull_request_id,
+      submitted_at,
+      state,
+      user_id
+    from macro
```
Comment (fivetran-jamie) [inline models/stg_github__pull_request_review.sql:38]: passthru

File: models/stg_github__repository.sql [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+        {% endif %}
+    from repository
+
+), fields as (
+
+    select 
+      id as repository_id,
+      full_name,
+      private as is_private
```
Comment (fivetran-jamie) [inline models/stg_github__repository.sql:35]: passthru

File: models/stg_github__requested_reviewer_history.sql [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+        {% endif %}
+    from requested_reviewer_history
+
+), fields as (
+
+    select 
+      pull_request_id,
+      created_at,
+      requested_id,
+      removed
```
Comment (fivetran-jamie) [inline models/stg_github__requested_reviewer_history.sql:36]: passthru

File: models/stg_github__user.sql [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+    from github_user
+
+), fields as (
+
+    select
+      id as user_id,
+      login as login_name,
+      name,
+      bio,
+      company
```
Comment (fivetran-jamie) [inline models/stg_github__user.sql:37]: passthru

**Comment by fivetran-joemarkiewicz** [comment]
@fivetran-jamie Thanks for looking through all those changes and calling out what needed to be updated. In this recent commit I made the versions to ```0.1.0``` as well as updated the typos and references to Intercom.

Additionally, I added the passthrough columns to each of the final models while also removing the ```!= []``` within the original pass through if statement as it is not necessary.

**Comment by fivetran-jamie** [comment]
looks good to me!

**Review by fivetran-jamie — APPROVED** [review]

**Pull request merged**